Introduce new user profile settings page

chrislessard opened the pull request:
This series of patches implements the design specification for the new user settings page, as described [here](https://github.com/openshiftio/openshift.io/issues/2662). It introduces four new sections : the user's Connected Accounts, Feature Opt-in preferences, Notification preferences and Resource usage overview. The work done here also solves https://github.com/openshiftio/openshift.io/issues/2659 

A few issues with the design specification - resulting from the fact that some features are not yet ready - came up and were discussed via mattermost. They include: 

1. It doesn't appear that OSIO can function in the context of this page unless an openshift account is linked, so there is no way to disable it. The associated button and tests based on this assumption have been dropped. There was no api to refresh openshift as far as I could find.
2. The feature flag portion of the spec involves dropdowns, but the information described in the dropdown is not yet available
3.  It doesn't appear to be the case that any of the notifications page can actually be toggled yet, so these have all been enabled by default and each button locked. 
4. A majority portion of the resources overview is not yet available, although @ebaron is working on it. I also had to make a miniature clone of the deployments service while @andrewazores refactors it (https://github.com/openshiftio/openshift.io/issues/2921).

Screenshots: 

Connected accounts: 

![2018-04-09-142716_1917x967_scrot](https://user-images.githubusercontent.com/17230733/38520316-6fe132aa-3c10-11e8-9b54-4bd9fe69a18f.png)

![2018-04-09-142744_1914x964_scrot](https://user-images.githubusercontent.com/17230733/38520315-6fd48b68-3c10-11e8-9d3c-e48c5c6e288e.png)

Feature opt-in:
![2018-04-09-142759_1917x967_scrot](https://user-images.githubusercontent.com/17230733/38520314-6fc57290-3c10-11e8-8352-4b151b8214a3.png)

![2018-04-09-142808_1909x962_scrot](https://user-images.githubusercontent.com/17230733/38520313-6fb770d2-3c10-11e8-8f52-15a60bea1eaa.png)

Notifications:
![2018-04-09-142817_1918x968_scrot](https://user-images.githubusercontent.com/17230733/38520312-6fab081a-3c10-11e8-9829-30dd1802f837.png)

Resources:

![2018-04-09-142835_1916x962_scrot](https://user-images.githubusercontent.com/17230733/38520311-6f9b44de-3c10-11e8-8194-d2dd72107966.png)

chrislessard commented:
@corinnekrych @catrobson please have a look at this when you get the chance !

mindreeper2420 commented on src/app/profile/settings/connected-accounts/connected-accounts.component.html:
```diff
@@ -0,0 +1,52 @@
+      <div class="card-pf">
+        <div class="card-pf-heading">
+          <div class="row account-content">
+            <div class="col-xs-6 col-sm-6 col-md-6">
```
No need to define this as being 6 columns for each size. If you set it to just `col-xs-6`, then it will fill 6 columns for all media queries.

mindreeper2420 commented on src/app/profile/settings/connected-accounts/connected-accounts.component.html:
```diff
@@ -0,0 +1,52 @@
+            <div class="col-xs-6 col-sm-6 col-md-6">
+              <h2 class="card-pf-title">Connected Accounts</h2>
+            </div>
+            <div class="col-xs-6 col-sm-6 col-md-6">
```
same here - just `col-xs-6`

mindreeper2420 commented on src/app/profile/settings/connected-accounts/connected-accounts.component.html:
```diff
@@ -0,0 +1,52 @@
+              <h2 class="card-pf-title">Connected Accounts</h2>
+            </div>
+            <div class="col-xs-6 col-sm-6 col-md-6">
+              <button class="btn btn-link pull-right" style="padding-right: 0" type="button" (click)="refreshServices()">
```
Please use the class `padding-right-0` rather than an inline style.

mindreeper2420 commented on src/app/profile/settings/connected-accounts/connected-accounts.component.html:
```diff
@@ -0,0 +1,52 @@
+        </div>
+        <div class="card-pf-body">
+          <div class="row account-content github-account-info">
+            <div class="col-xs-2 col-sm-2 col-md-2">
```
just define `col-xs-2`

mindreeper2420 commented on src/app/profile/settings/connected-accounts/connected-accounts.component.html:
```diff
@@ -0,0 +1,52 @@
+            <div class="col-xs-2 col-sm-2 col-md-2">
+              <i class="fa fa-github" style="font-size: 3em"></i>
+            </div>
+            <div class="account-text-box col-xs-6 col-sm-6 col-md-6">
```
`col-xs-6`

mindreeper2420 commented on src/app/profile/settings/connected-accounts/connected-accounts.component.html:
```diff
@@ -0,0 +1,52 @@
+              <span class="word-wrap" *ngIf="gitHubLinked; else errorName">{{gitHubUserName}}</span>
+              <ng-template #errorName><span>{{gitHubError}}</span></ng-template>
+            </div>
+            <div class="col-xs-4 col-sm-4 col-md-4">
```
`col-xs-4`

mindreeper2420 commented on src/app/profile/settings/connected-accounts/connected-accounts.component.html:
```diff
@@ -0,0 +1,52 @@
+            </div>
+          </div>
+          <div class="row account-content">
+            <div class="col-xs-2 col-sm-2 col-md-2">
```
`col-xs-2`

mindreeper2420 commented on src/app/profile/settings/connected-accounts/connected-accounts.component.html:
```diff
@@ -0,0 +1,52 @@
+          </div>
+          <div class="row account-content">
+            <div class="col-xs-2 col-sm-2 col-md-2">
+              <img class="icon" height="36px" width="42px" src="../../../../assets/images/openshift_icon.svg">
```
If you're defining a class here, please add the height and width dimensions to that, rather than as an inline definition.

mindreeper2420 commented on src/app/profile/settings/connected-accounts/connected-accounts.component.html:
```diff
@@ -0,0 +1,52 @@
+              <b>GitHub</b>
+              <br>
+              <span class="word-wrap" *ngIf="gitHubLinked; else errorName">{{gitHubUserName}}</span>
+              <ng-template #errorName><span>{{gitHubError}}</span></ng-template>
```
`<span>{{gitHubError}}</span>` should be a new line for readability

mindreeper2420 commented on src/app/profile/settings/connected-accounts/connected-accounts.component.html:
```diff
@@ -0,0 +1,52 @@
+            <div class="col-xs-2 col-sm-2 col-md-2">
+              <img class="icon" height="36px" width="42px" src="../../../../assets/images/openshift_icon.svg">
+            </div>
+            <div class="account-text-box col-xs-6 col-sm-6 col-md-6">
```
`col-xs-6`

mindreeper2420 commented on src/app/profile/settings/connected-accounts/connected-accounts.component.html:
```diff
@@ -0,0 +1,52 @@
+            </div>
+            <div class="account-text-box col-xs-6 col-sm-6 col-md-6">
+              <b>OpenShift</b>
+              <br>
```
avoid the use of `<br>` if possible - adding a class and bottom margin to `<b>OpenShift</b>` would suffice.

mindreeper2420 commented on src/app/profile/settings/connected-accounts/connected-accounts.component.html:
```diff
@@ -0,0 +1,52 @@
+              <img class="icon" height="36px" width="42px" src="../../../../assets/images/openshift_icon.svg">
+            </div>
+            <div class="account-text-box col-xs-6 col-sm-6 col-md-6">
+              <b>OpenShift</b>
```
Use a class for defining the font-weight of this title, rather than `<b>`

mindreeper2420 commented on src/app/profile/settings/connected-accounts/connected-accounts.component.html:
```diff
@@ -0,0 +1,52 @@
+              <br>
+              <span class="word-wrap">{{openShiftUserName}}</span>
+            </div>
+            <div class="col-xs-4 col-sm-4 col-md-4"></div>
```
`col-xs-4`

mindreeper2420 commented on src/app/profile/settings/connected-accounts/connected-accounts.component.html:
```diff
@@ -0,0 +1,52 @@
+        </div>
+      </div>
+    </div>
+    <div class="col-xs-2 col-sm-8 col-md-8"></div>
```
For this, you would only need `col-xs-2 col-sm-8`.

mindreeper2420 commented on src/app/profile/settings/feature-opt-in/feature-opt-in.component.html:
```diff
@@ -0,0 +1,95 @@
+<div class="container-fluid container-cards-pf">
+  <div class="col-xs-10 col-sm-6 col-md-6">
```
You only need to define `col-xs-10 col-sm-6`.

mindreeper2420 commented on src/app/profile/settings/feature-opt-in/feature-opt-in.component.html:
```diff
@@ -0,0 +1,95 @@
+          </div>
+          <div class="list-view-pf-main-info">
+            <div class="list-view-pf-left">
+              <span class="fa list-view-pf-icon-sm" style="font-family: 'Open Sans'; font-size:18px; line-height: 1.3"> ß </span>
```
Please do not use inline styles - create a class and apply these styles to that class.

mindreeper2420 commented on src/app/profile/settings/feature-opt-in/feature-opt-in.component.html:
```diff
@@ -0,0 +1,95 @@
+                  Internal Experimental Features
+                </div>
+                <div class="list-group-item-text">
+                  These experimetnal features are released to internal employees only.
```
Spelling - update to 'experimental'

mindreeper2420 commented on src/app/profile/settings/feature-opt-in/feature-opt-in.component.html:
```diff
@@ -0,0 +1,95 @@
+      </div>
+    </div>
+  </div>
+  <div class="col-xs-2 col-sm-6 col-md-6"></div>
```
What is this empty `<div>` for?

chrislessard replied:
I thought you had to introduce empty divs so that all the columns added up to twelve, I see that it's not necessary !

mindreeper2420 commented on src/app/profile/settings/notifications/notifications.component.html:
```diff
@@ -0,0 +1,91 @@
+<div class="container-fluid container-cards-pf">
+  <div class="row-cards-pf">
+    <div class="col-xs-10 col-sm-6 col-md-6">
```
Only needs `col-xs-10 col-sm-6` and not the addition `col-md-6`

mindreeper2420 commented on src/app/profile/settings/notifications/notifications.component.html:
```diff
@@ -0,0 +1,91 @@
+      <div class="card-pf">
+        <div class="card-pf-heading">
+          <div class="row notification-option-content">
+            <div class="col-xs-6 col-sm-6 col-md-6">
```
Only needs `col-xs-6`

mindreeper2420 commented on src/app/profile/settings/notifications/notifications.component.html:
```diff
@@ -0,0 +1,91 @@
+        </div>
+        <div class="card-pf-body">
+          <div class="row github-notification-option-info">
+            <div class="col-xs-1 col-sm-1 col-md-1">
```
Only needs `col-xs-1`

mindreeper2420 commented on src/app/profile/settings/notifications/notifications.component.html:
```diff
@@ -0,0 +1,91 @@
+            <div class="col-xs-1 col-sm-1 col-md-1">
+              <input type="checkbox" [disabled]="true" [checked]="true">
+            </div>
+            <div class="notification-option-text-box col-xs-10 col-sm-10 col-md-10">
```
Only needs `col-xs-10`

mindreeper2420 commented on src/app/profile/settings/notifications/notifications.component.html:
```diff
@@ -0,0 +1,91 @@
+            </div>
+          </div>
+          <div class="row github-notification-option-info">
+            <div class="col-xs-1 col-sm-1 col-md-1">
```
Only `col-xs-1` is needed

mindreeper2420 commented on src/app/profile/settings/notifications/notifications.component.html:
```diff
@@ -0,0 +1,91 @@
+            <div class="col-xs-1 col-sm-1 col-md-1">
+                <input type="checkbox" [disabled]="true" [checked]="true">
+              </div>
+            <div class="notification-option-text-box col-xs-10 col-sm-10 col-md-10">
```
Only `col-xs-10` is needed

mindreeper2420 commented on src/app/profile/settings/notifications/notifications.component.html:
```diff
@@ -0,0 +1,91 @@
+              </div>
+            <div class="notification-option-text-box col-xs-10 col-sm-10 col-md-10">
+              <b>Email Notifications</b>
+              <br>
```
It is preferred to not use `<br>` and apply a margin to the bottom of "Email Notifications".

mindreeper2420 commented on src/app/profile/settings/notifications/notifications.component.html:
```diff
@@ -0,0 +1,91 @@
+        </div>
+      </div>
+    </div>
+    <div class="col-xs-2 col-sm-6 col-md-6"></div>
```
What is this empty `<div>` for?

mindreeper2420 commented on src/app/profile/settings/notifications/notifications.component.html:
```diff
@@ -0,0 +1,91 @@
+</div>
+<div class="container-fluid container-cards-pf">
+  <div class="row-cards-pf">
+    <div class="col-xs-10 col-sm-6 col-md-6">
```
Only `col-xs-10 col-sm-6` is needed.

mindreeper2420 commented on src/app/profile/settings/notifications/notifications.component.html:
```diff
@@ -0,0 +1,91 @@
+      <div class="card-pf">
+        <div class="card-pf-heading">
+          <div class="row notification-option-content">
+            <div class="col-xs-6 col-sm-6 col-md-6">
```
Only `col-xs-6` is needed.

mindreeper2420 commented on src/app/profile/settings/notifications/notifications.component.html:
```diff
@@ -0,0 +1,91 @@
+        </div>
+        <div class="card-pf-body">
+          <div class="row github-notification-option-info">
+            <div class="col-xs-1 col-sm-1 col-md-1">
```
Just `col-xs-1`

mindreeper2420 commented on src/app/profile/settings/notifications/notifications.component.html:
```diff
@@ -0,0 +1,91 @@
+            <div class="col-xs-1 col-sm-1 col-md-1">
+                <input type="checkbox" [disabled]="true" [checked]="true">
+              </div>
+            <div class="notification-option-text-box col-xs-11 col-sm-11 col-md-11">
```
`col-xs-11` is all that is needed

mindreeper2420 commented on src/app/profile/settings/notifications/notifications.component.html:
```diff
@@ -0,0 +1,91 @@
+              </div>
+            <div class="notification-option-text-box col-xs-11 col-sm-11 col-md-11">
+              <b>Assigned to an issue</b>
+              <br>
```
Preferred method is to use a margin on the element above - "Assigned to an issue".

mindreeper2420 commented on src/app/profile/settings/notifications/notifications.component.html:
```diff
@@ -0,0 +1,91 @@
+            </div>
+          </div>
+          <div class="row github-notification-option-info">
+            <div class="col-xs-1 col-sm-1 col-md-1">
```
Only `col-xs-1` is needed.

mindreeper2420 commented on src/app/profile/settings/notifications/notifications.component.html:
```diff
@@ -0,0 +1,91 @@
+            <div class="col-xs-1 col-sm-1 col-md-1">
+                <input type="checkbox" [disabled]="true" [checked]="true">
+              </div>
+            <div class="notification-option-text-box col-xs-10 col-sm-10 col-md-10">
```
Remove `col-md-10`

mindreeper2420 commented on src/app/profile/settings/notifications/notifications.component.html:
```diff
@@ -0,0 +1,91 @@
+              </div>
+            <div class="notification-option-text-box col-xs-10 col-sm-10 col-md-10">
+              <b>Status changes</b>
+              <br>
```
Replace with a bottom margin on the "Status changes" element.

mindreeper2420 commented on src/app/profile/settings/notifications/notifications.component.html:
```diff
@@ -0,0 +1,91 @@
+            </div>
+          </div>
+          <div class="row github-notification-option-info">
+            <div class="col-xs-1 col-sm-1 col-md-1">
```
Use `col-xs-1`

mindreeper2420 commented on src/app/profile/settings/notifications/notifications.component.html:
```diff
@@ -0,0 +1,91 @@
+            <div class="col-xs-1 col-sm-1 col-md-1">
+                <input type="checkbox" [disabled]="true" [checked]="true">
+              </div>
+            <div class="notification-option-text-box col-xs-10 col-sm-10 col-md-10">
```
`col-xs-10`

mindreeper2420 commented on src/app/profile/settings/notifications/notifications.component.html:
```diff
@@ -0,0 +1,91 @@
+                <input type="checkbox" [disabled]="true" [checked]="true">
+              </div>
+            <div class="notification-option-text-box col-xs-10 col-sm-10 col-md-10">
+              <b>Comments</b>
```
Remove `<br>` and replace with a bottom margin.

mindreeper2420 commented on src/app/profile/settings/notifications/notifications.component.html:
```diff
@@ -0,0 +1,91 @@
+        </div>
+      </div>
+    </div>
+    <div class="col-xs-2 col-sm-6 col-md-6">
```
You shouldn't use an empty `<div>` with a `<br>` inside of it.

mindreeper2420 commented on src/app/profile/settings/settings.component.less:
```diff
@@ -0,0 +1,45 @@
+@import (reference) '../../../assets/stylesheets/shared/osio.less';
+
```
Replace all hex colors with PatternFly variables. Refer to this page for color variables: http://www.patternfly.org/styles/color-palette/. Everything is preface `@color-` followed by the color's name.

mindreeper2420 commented on src/app/profile/settings/settings.component.less:
```diff
@@ -0,0 +1,45 @@
+  background-color: #393F44;
+  color: #ffffff;
+  height: 60px;
+  padding: 18px 15px 15px 25px;
```
Simplify this definition.

mindreeper2420 commented on src/app/profile/settings/settings.component.less:
```diff
@@ -0,0 +1,45 @@
+  padding: 18px 15px 15px 25px;
+}
+
+#settings-title-nav h2, h3 {
```
You should use a class, rather than ID's and subsequent targeting of HTML elements.

mindreeper2420 commented on src/app/profile/settings/settings.component.less:
```diff
@@ -0,0 +1,45 @@
+}
+
+#settings-options-nav {
+  padding: 10px 0px 10px 0px;
```
Simplify to `padding: 10px 0;`

alexeykazakov commented:
How do I re-connect GitHub or OpenShift Online accounts?
What does the "Refresh" button do? Is it for re-connecting accounts? If so, then why one button instead of one button per account. Why it's called "Refresh"?
I asked the same questions in https://redhat.invisionapp.com/share/FWFGF7VXMB8#/screens/282415824/comments but no answers.

chrislessard commented:
@mceledonia @alexeykazakov keeping with the discussion about connected accounts, I'm a bit confused by the most recent image Michael put up. There doesn't seem to be any way of unlinking the user's openshift account, so should that button even be there ?

alexeykazakov commented:
@chrislessard backend supports disconnecting OpenShift accounts but it was decided to not add it to UI because OSO link is required. However we can allow disconnecting GitHub because New App Wizard and other places detects missing GH links and initiate GH account linking.
So, don't add Disconnect button for OSO. It should be available for GitHub only.
Refresh/Reconnect buttons/icons should be available for both.

mceledonia commented:
@chrislessard That was my fault I used an outdated version of the screen to display how the reconnect icon should be treated. I will update the image in that post.

corinnekrych commented:
@chrislessard same issue than Jie on `npm start`

chrislessard commented:
@corinnekrych @jiekang i've addressed adams comments, so there shouldn't be any stylint issues anymore. I'm getting to the other requests now.

corinnekrych commented:
LGTM

jiekang commented:
@chrislessard @mceledonia 

1. Currently nothing happens when mousing over the individual sub-tabs, Connected Accounts, Features Opt-In, Notifications and Resources. I feel like there should be some kind of graphical change when hovering with the mouse to indicate clickability. E.g. a highlight occurs when hovering the Analyze, Plan and Create tabs, or the Create/Planner sub tabs.
2. The tab system is currently graphically inconsistent with the 'Create' and 'Planner' tabs. Is this a concern? Are there plans to update the other pages instead? See images below.

![create-tab](https://user-images.githubusercontent.com/5430520/38678674-b0cf9d6c-3e2f-11e8-9599-2a40cda4f22d.png)

![plan-tab](https://user-images.githubusercontent.com/5430520/38678686-b8541eb4-3e2f-11e8-959a-1f98a093da38.png)

![settings-tab](https://user-images.githubusercontent.com/5430520/38678679-b27e46fe-3e2f-11e8-8e91-66d197e712d8.png)

jiekang commented:
3. For the 'Features Opt-In' page, mousing over each selection highlights the card, suggesting it can be clicked to select that feature level. However, only the checkbox is clickable. I'd prefer the entire card be clickable.

4. For the 'Notifications' page, my understanding is the feature isn't available so the toggles are locked. I think they should be in a disabled state, e.g. greyed out in some fashion. Alternatively, there needs to be some graphic or text on that page that communicates to the user that the notification features are disabled.

jiekang commented:
6. The refresh flows for GitHub and Openshift.io need to be examined.

When running a local instance against production, refreshing GitHub redirects me to the settings page with a query parameter. Why is this redirect necessary?

When running a local instance against production, refreshing Openshift leads to https://api.starter-us-east-2.openshift.com/oauth/authorize with message 'An authentication error occurred'. My account is temporarily inaccessible in the browser instance, even via openshift.io. 

This suggests, if this is expected, a warning needs to be in place when refreshing the Openshift link that the account will be temporarily unavailable while relinking occurs. If this is unexpected, we need to review what should be happening and why it's not. Hopefully it's expected so there's less work :)

mceledonia commented:
@chrislessard @jiekang 

1. Yes we should definitely include a hover state. I think if you make the text change color to the same blue color it changes to when it's selected, that should be great. That still separates it from the selected state because of the underline. 

2. We should update the styling that you see to the same as the user settings (third image) style. The blue text is more difficult to read over the dark background, which is why I think we should change to the lighter background seen in that image and have them all be consistent.

3. Agreed, let's make the whole card clickable except for any separate actions like the expand button.

4. ...

alexeykazakov commented:
> By any chance, do you know what's supposed to occur when 'Refreshing' the Openshift link?

OSIO redirects user to OpenShift Online. If user is already logged in to OSO then the user redirects back to OSIO. If not logged-in then OSO will redirects to developers.redhat.com for logging and then back to OSO and OSIO.
There should not be any temporal or permanent inaccessibility of the account during this.

chrislessard commented:
@corinnekrych thank you !

@mindreeper2420 I've addressed all of your changes, thanks for those catches.

@jiekang @mceledonia 
Going through the list:
1. I've added highlighting to the hovered links, no underline.
2. That's fine with me, but it's work for a separate pr. I'll open an associated issue for it.
3. I've made the whole div clickable.
4. I've added a basic slider than Adam sent to me. It isn't toggle-able as it stands (i've commented out the relevant css that makes it so), this should be extracted out into a separate global component but again, that's work for a separate pr. 
5. @mceledonia I need your approval on what to do here. I think dropping the arrows for now makes sense.
6. I think these things need to be refactored in general, but since this PR is meant to implement UX, I'd rather it be done elsewhere in a place where the conversation can be more contained. @jiekang , I believe the reason for this temporary logging out can be explained by the current implementation, which I just moved over, that can be seen here https://github.com/fabric8-ui/fabric8-ui/blob/master/src/app/profile/overview/connected-accounts/connected-accounts.component.ts#L85

This certainly doesn't seem right given what @alexeykazakov was describing, but again, I'm not entirely sure what fixing it entails and would prefer a separate issue be opened with people who can help me out, if I'm to be the one to fix it.

mindreeper2420 commented on src/app/profile/settings/connected-accounts/connected-accounts.component.html:
```diff
@@ -0,0 +1,52 @@
+<div class="container-fluid container-cards-pf">
+  <div class="row-cards-pf accounts-component">
+    <div class="col-xs-10 col-sm-4 col-md-4">
```
You can remove the additional `col-md-4`, as `col-sm-4` defines the same column width.

mindreeper2420 commented on src/app/profile/settings/connected-accounts/connected-accounts.component.html:
```diff
@@ -0,0 +1,52 @@
+        </div>
+      </div>
+    </div>
+    <div class="col-xs-2 col-sm-8"></div>
```
Empty `<div>` not necessary.

mindreeper2420 commented on src/app/profile/settings/connected-accounts/connected-accounts.component.html:
```diff
@@ -0,0 +1,52 @@
+                <span class="fa fa-refresh"></span>
+              </button>
+            </div>
+            <div class="col-xs-4"></div>
```
Empty `<div>` not necessary.

mindreeper2420 commented on src/app/profile/settings/feature-opt-in/feature-opt-in.component.less:
```diff
@@ -0,0 +1,25 @@
+@import (reference) '../../../../assets/stylesheets/shared/osio.less';
+
```
Not sure why I didn't catch this before, but why define the `font-family` for each element?

chrislessard replied:
fixed !

mindreeper2420 commented on src/app/profile/settings/feature-opt-in/feature-opt-in.component.less:
```diff
@@ -0,0 +1,25 @@
+.feature-title {
+  font-weight: bold;
+  font-family: 'Open Sans';
+  font-size: 14px;
```
Use the variable for the base font size `@font-size-base`

mindreeper2420 commented on src/app/profile/settings/feature-opt-in/feature-opt-in.component.less:
```diff
@@ -0,0 +1,25 @@
+.feature-description {
+  margin-bottom: 0;
+  font-family: 'Open Sans';
+  font-size: 12px;
```
Set as an `em` of the 14px default.

mindreeper2420 commented on src/app/profile/settings/notifications/notifications.component.html:
```diff
@@ -0,0 +1,83 @@
+            <div class="col-xs-1">
+                <toggle-bar [id]="2"></toggle-bar>
+              </div>
+            <div class="col-xs-11 col-sm-11 col-md-11">
```
Simplify to `<div class="col-xs-11">`

mindreeper2420 commented on src/app/profile/settings/notifications/notifications.component.less:
```diff
@@ -0,0 +1,15 @@
+.notification-entry-title {
+  display: block;
+  margin-bottom: 3px;
+  font-weight: bold;
```
Update to use `font-weight: 700`.

mindreeper2420 commented on src/app/profile/settings/notifications/toggle-bar.component.less:
```diff
@@ -0,0 +1,66 @@
+      position: relative;
+      display: block;
+      content: '';
+      width: 50%;
```
You have percentages for width and height here, but are using `em` above - can you have these the same value type?

mindreeper2420 commented on src/app/profile/settings/notifications/toggle-bar.component.less:
```diff
@@ -0,0 +1,66 @@
+      display: none;
+    }
+  }
+  // &:checked + .toggle-btn:after {
```
Remove commented out code.

mindreeper2420 commented on src/app/profile/settings/notifications/toggle-bar.component.less:
```diff
@@ -0,0 +1,66 @@
+    border-radius: 1px;
+    border: 1px solid #bbb;
+    padding: 0;
+    //transition: all .4s ease;
```
Remove commented out code.

mindreeper2420 commented on src/app/profile/settings/notifications/toggle-bar.component.less:
```diff
@@ -0,0 +1,66 @@
+  + .toggle-btn {
+    background: @color-pf-black-200;
+    border-radius: 1px;
+    border: 1px solid #bbb;
```
Use PatternFly variable instead of hex code.

mindreeper2420 commented on src/app/profile/settings/notifications/toggle-bar.component.less:
```diff
@@ -0,0 +1,66 @@
+      box-shadow: 0 0 2px rgba(3, 3, 3, .4);
+      background-image: linear-gradient(to bottom, @color-pf-black-100  0, @color-pf-black-200  100%);
+      background-repeat: repeat-x;
+      //transition: all .3s ease;
```
Remove commented out code.

chrislessard replied:
I did this on the presumption that we would eventually want transitions to be activated, at which point we would un-comment. Is this undesirable ?

mindreeper2420 replied:
It's best to remove the unused code, even if it'll be implemented at a later date. We can standardize transitions, and then just use a variable for easier updates.

mindreeper2420 commented on src/app/profile/settings/notifications/toggle-bar.component.less:
```diff
@@ -0,0 +1,66 @@
+      z-index: 9;
+    }
+  }
+  // &:checked + .toggle-btn {
```
Remove commented out code.

mindreeper2420 commented on src/app/profile/settings/resources/resources.component.html:
```diff
@@ -0,0 +1,50 @@
+  <div class="col-xs-6">
+    <div class="row resource-header-colour list-group-item">
+      <div>
+        <div class="col-sm-6">
```
If you are using `col-xs-` as the parent column definition, please continue that into the child elements (change `col-sm-` to `col-xs-`).

mindreeper2420 commented on src/app/profile/settings/resources/resources.component.html:
```diff
@@ -0,0 +1,50 @@
+    </div>
+  </div>
+</div>
+<div class="col-xs-6"></div>
```
Remove empty `<div>`

mindreeper2420 commented on src/app/profile/settings/resources/resources.component.less:
```diff
@@ -0,0 +1,28 @@
+}
+
+.resources-full-capacity {
+  background-color: #f9e5e5;
```
Use PatternFly color variable.

chrislessard replied:
It doesn't seem like there is one ? The colour is trying to match the background on the entry when full capacity is reached, as seen here https://redhat.invisionapp.com/share/FWFGF7VXMB8#/screens/282415828

mceledonia commented:
@chrislessard Regarding the down arrows, let's drop them for now as you suggest, but be sure to keep in mind we will want to add them back later.

jiekang commented:
@chrislessard for the items that are appropriate as separate tasks, can you open issues and link them for me? Thanks! I believe it's just points 2 & 6.

jiekang commented:
The connected accounts page has weird spacing for me now:

![connected-accounts](https://user-images.githubusercontent.com/5430520/38832533-d8df4e6a-4190-11e8-8b24-a8b7aad4a591.png)

I have an application deployed in stage taking up resources (as shown in Deployments page) but the resources tab for the Settings page still shows nothing taken. Is the functionality not there?

chrislessard commented:
@jiekang Both have been fixed. (1) Was a result of making an error when addressing @mindreeper2420 changes and (2) was a result of an object memory address related bug.

chrislessard commented:
@jiekang as for the follow up issues, i've gone ahead and created:

(1) https://github.com/openshiftio/openshift.io/issues/3163
(2) https://github.com/openshiftio/openshift.io/issues/3164

jiekang commented:
Code looks okay. Functionality looks okay. If anything minor comes up in those areas it can be for a follow-up PR.

jiekang commented on src/app/profile/settings/resources/resources.component.ts:
```diff
@@ -0,0 +1,103 @@
+    let loggedInUser = this.userService.currentLoggedInUser.attributes.username;
+
+    this.subscriptions.push(this.resourceService.getEnvironments().subscribe(
+      (projNames) => {
```
I didn't look at this closely till now. This logic should be moved to the service. The view should just receive the object to be displayed.

chrislessard commented:
@jiekang I've moved all the component logic to the service, and migrated the tests there as well. Please have a look !

jiekang commented on src/app/profile/settings/services/resource.service.ts:
```diff
@@ -0,0 +1,91 @@
+    private userService: UserService
+  ) {}
+
+  public getEnvironmentsWithScaleAndIcon(): Observable<UsageSeverityEnvironmentStat[]> {
```
Please open an issue to refactor this service class, or update one if the issue already exists. I'm surprised there's no test that calls `getEnvironmentsWithScaleAndIcon()`

If the other functions are indeed public, to be used elsewhere I'd like them to explicitly be public. If they're only intended as helpers for this public function, they should be made private.

fabric8cd commented:
@chrislessard fabric8/fabric8-ui:SNAPSHOT-PR-2711-15 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2711-fabric8-ui.badger.fabric8.io

joshuawilson commented:
This should not have been merged, it has un-filtered changes to the UI. Docs have been impacted. 
We either have to wait for Docs to update or we have to revert it.

joshuawilson commented:
I was following it. I could see working being requested and done. If it doesn't have the feature flags then we need to tell docs about it. Then we can schedule the merge with them. 
If it had been squashed it would be much easier to revert it. I don't think it will be as easy now.
